[Ubuntu 20.04] Setup fails when provisioning VMs due to DNS error

My setup: base vanilla server with just needed packaged installed

vars/k8s_cluster.yml changes: **container runtime containerd** and **kube version 1.21**


```
Error

PLAY [Check connection and set facts] **********************************************

TASK [Wait 600 seconds for target connection to become reachable/usable] ***********
fatal: [k8s-test-master-0.k8s.test]: FAILED! => {"changed": false, "elapsed": 600, "msg": "timed out waiting for ping module test: Failed to connect to the host via ssh: ssh: Could not resolve hostname k8s-test-master-0.k8s.test: Name or service not known"}
fatal: [k8s-test-master-2.k8s.test]: FAILED! => {"changed": false, "elapsed": 600, "msg": "timed out waiting for ping module test: Failed to connect to the host via ssh: ssh: Could not resolve hostname k8s-test-master-2.k8s.test: Name or service not known"}
fatal: [k8s-test-worker-0.k8s.test]: FAILED! => {"changed": false, "elapsed": 600, "msg": "timed out waiting for ping module test: Failed to connect to the host via ssh: ssh: Could not resolve hostname k8s-test-worker-0.k8s.test: Name or service not known"}
fatal: [k8s-test-master-1.k8s.test]: FAILED! => {"changed": false, "elapsed": 600, "msg": "timed out waiting for ping module test: Failed to connect to the host via ssh: ssh: Could not resolve hostname k8s-test-master-1.k8s.test: Name or service not known"}

PLAY RECAP *************************************************************************
k8s-test-master-0.k8s.test : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0
k8s-test-master-1.k8s.test : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0
k8s-test-master-2.k8s.test : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0
k8s-test-worker-0.k8s.test : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0
localhost                  : ok=36   changed=24   unreachable=0    failed=0    skipped=16   rescued=0    ignored=0

make: *** [Makefile:13: create] Error 2
```

DNS resolution from host is in fact broken
DNS resolution from inside the vms works


More Infos

```
virsh list --all
 Id   Name                State
-----------------------------------
 1    k8s-test-master-2   running
 2    k8s-test-master-1   running
 3    k8s-test-master-0   running
 4    k8s-test-worker-0   running
```

```
 Name       MAC address          Protocol     Address
-------------------------------------------------------------------------------
 vnet0      52:54:00:b6:34:b9    ipv4         192.168.200.143/24

 Name       MAC address          Protocol     Address
-------------------------------------------------------------------------------
 vnet2      52:54:00:f6:86:8c    ipv4         192.168.200.152/24

 Name       MAC address          Protocol     Address
-------------------------------------------------------------------------------
 vnet1      52:54:00:b2:12:c5    ipv4         192.168.200.65/24

 Name       MAC address          Protocol     Address
-------------------------------------------------------------------------------
 vnet3      52:54:00:1e:93:48    ipv4         192.168.200.81/24
```

```
dig k8s-test-master-0.k8s.test

; <<>> DiG 9.16.1-Ubuntu <<>> k8s-test-master-0.k8s.test
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NXDOMAIN, id: 40048
;; flags: qr rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 1, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 512
;; QUESTION SECTION:
;k8s-test-master-0.k8s.test.	IN	A

;; AUTHORITY SECTION:
.			42	IN	SOA	a.root-servers.net. nstld.verisign-grs.com. 2021101200 1800 900 604800 86400

;; Query time: 3 msec
;; SERVER: 147.75.207.207#53(147.75.207.207)
;; WHEN: Tue Oct 12 09:09:57 UTC 2021
;; MSG SIZE  rcvd: 130
```


```

resolvectl query k8s-test-master-0.k8s.test
k8s-test-master-0.k8s.test: 192.168.200.65     -- link: k8s-test

-- Information acquired via protocol DNS in 1.0ms.
-- Data is authenticated: no
```

--- a/03_provision_libvirt_resources.yml
+++ b/03_provision_libvirt_resources.yml
@@ -20,8 +20,8 @@
 
     - name: Take care of systemd-resolved on F33 and Ubuntu hosts
       when:
-        (ansible_distribution == 'Fedora' and ansible_distribution_major_version | int < 33) or
-        (ansible_distribution == 'Ubuntu')
+        (ansible_distribution == 'Fedora' and ansible_distribution_major_version | int > 33) or
+        (ansible_distribution == 'Ubuntu' and ansible_distribution_major_version | int > 18)
       block:
         - name: Ensure systemd-resolved config dir is present
           ansible.builtin.file:

--- a/03_provision_libvirt_resources.yml
+++ b/03_provision_libvirt_resources.yml
@@ -21,7 +21,7 @@
     - name: Take care of systemd-resolved on F33 and Ubuntu hosts
       when:
         (ansible_distribution == 'Fedora' and ansible_distribution_major_version | int < 33) or
-        (ansible_distribution == 'Ubuntu' and ansible_distribution_major_version | int < 18)
+        (ansible_distribution == 'Ubuntu')
       block:
         - name: Ensure systemd-resolved config dir is present
           ansible.builtin.file:

